ui/client/documents/index.js

```diff
@@ -380,7 +355,17 @@ export const ParagraphTile = withStyles((theme) => ({
             <dt>
               <Chip classes={{root: classes.squareChip, label: classes.chipLabel}} label="text" />
             </dt>
-            <dd>{highlightText(paragraph.text, highlights)}</dd>
+            <dd>{(highlights || calculateHighlightTargets(paragraph.text, query))
```

Comment:
Made it so that UI first highlights while awaiting for extended semantic highlighter results.